components/suggest/src/db.rs

```diff
@@ -125,6 +125,10 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
+        if let Some(suggestion) = yelp::fetch_suggestion(self.conn, query)? {
```

Comment:
Just a note for future refactors. We might want to refactor this to move it to the query execution block down below. Otherwise, all the calls regardless of the interested providers will unconditionally call `yelp::fetch_suggestion()`.